src/e2e/app/common/src/e2e_harness.rs

```diff
@@ -0,0 +1,186 @@
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::future::Future;
+
+use kamu_cli_e2e_common::KamuApiServerClient;
+use kamu_node_puppet::extensions::KamuNodePuppetExt;
+use kamu_node_puppet::{KamuNodePuppet, NewWorkspaceOptions};
+use regex::Regex;
+use sqlx::{PgPool, SqlitePool};
+
+use crate::{api_flight_sql_e2e_test, api_server_e2e_test, KamuFlightSQLClient};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Default)]
+pub struct KamuNodeApiServerHarnessOptions {
+    env_vars: Vec<(String, String)>,
+    kamu_api_server_config: Option<String>,
+}
+
+impl KamuNodeApiServerHarnessOptions {
+    pub fn with_kamu_config(mut self, content: &str) -> Self {
+        self.kamu_api_server_config = Some(content.into());
+
+        self
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct KamuNodeApiServerHarness {
+    options: KamuNodeApiServerHarnessOptions,
+}
+
+impl KamuNodeApiServerHarness {
+    pub fn postgres(pg_pool: &PgPool, options: KamuNodeApiServerHarnessOptions) -> Self {
+        let db = pg_pool.connect_options();
+        let kamu_config = indoc::formatdoc!(
+            r#"
+            database:
+                provider: postgres
+                host: {host}
+                credentialsPolicy:
+                    source:
+                        kind: rawPassword
+                        userName: {user}
+                        rawPassword: {password}
+                databaseName: {database}
+            "#,
+            host = db.get_host(),
+            user = db.get_username(),
+            password = db.get_username(), // It's intended: password is same as user for tests
+            database = db.get_database().unwrap(),
+        );
+
+        Self::new(options, Some(kamu_config))
+    }
+
+    pub fn sqlite(sqlite_pool: &SqlitePool, options: KamuNodeApiServerHarnessOptions) -> Self {
+        // Ugly way to get the path as the settings have a not-so-good signature:
+        // SqliteConnectOptions::get_filename(self) -> Cow<'static, Path>
+        //                                    ^^^^
+        // Arc<T> + consuming = bad combo
+        let database_path = {
+            let re = Regex::new(r#"filename: "(.*)""#).unwrap();
+            let connect_options = format!("{:#?}", sqlite_pool.connect_options());
+            let re_groups = re.captures(connect_options.as_str()).unwrap();
+            let relative_path = re_groups[1].to_string();
+
+            std::fs::canonicalize(relative_path)
+                .unwrap()
+                .display()
+                .to_string()
+        };
+
+        let kamu_config = indoc::formatdoc!(
```

Comment:
`let kamu_api_server_config = `

Reply:
General file comment: please just search for `kamu_config` and update all locations